spark/v4.0/spark/src/test/java/org/apache/iceberg/spark/source/TestSparkReaderWithBloomFilter.java

```diff
@@ -248,91 +248,90 @@ protected void dropTable(String name) {
-    factory.set(
+    writeProperties.put(
         PARQUET_BLOOM_FILTER_COLUMN_ENABLED_PREFIX + "id_float",
         Boolean.toString(useBloomFilterCol4));
     boolean useBloomFilterCol5 =
         PropertyUtil.propertyAsBoolean(
             table.properties(), PARQUET_BLOOM_FILTER_COLUMN_ENABLED_PREFIX + "id_string", false);
-    factory.set(
+    writeProperties.put(
         PARQUET_BLOOM_FILTER_COLUMN_ENABLED_PREFIX + "id_string",
         Boolean.toString(useBloomFilterCol5));
     boolean useBloomFilterCol6 =
         PropertyUtil.propertyAsBoolean(
             table.properties(), PARQUET_BLOOM_FILTER_COLUMN_ENABLED_PREFIX + "id_boolean", false);
-    factory.set(
+    writeProperties.put(
         PARQUET_BLOOM_FILTER_COLUMN_ENABLED_PREFIX + "id_boolean",
         Boolean.toString(useBloomFilterCol6));
     boolean useBloomFilterCol7 =
         PropertyUtil.propertyAsBoolean(
             table.properties(), PARQUET_BLOOM_FILTER_COLUMN_ENABLED_PREFIX + "id_date", false);
-    factory.set(
+    writeProperties.put(
         PARQUET_BLOOM_FILTER_COLUMN_ENABLED_PREFIX + "id_date",
         Boolean.toString(useBloomFilterCol7));
     boolean useBloomFilterCol8 =
         PropertyUtil.propertyAsBoolean(
             table.properties(),
             PARQUET_BLOOM_FILTER_COLUMN_ENABLED_PREFIX + "id_int_decimal",
             false);
-    factory.set(
+    writeProperties.put(
         PARQUET_BLOOM_FILTER_COLUMN_ENABLED_PREFIX + "id_int_decimal",
         Boolean.toString(useBloomFilterCol8));
     boolean useBloomFilterCol9 =
         PropertyUtil.propertyAsBoolean(
             table.properties(),
             PARQUET_BLOOM_FILTER_COLUMN_ENABLED_PREFIX + "id_long_decimal",
             false);
-    factory.set(
+    writeProperties.put(
         PARQUET_BLOOM_FILTER_COLUMN_ENABLED_PREFIX + "id_long_decimal",
         Boolean.toString(useBloomFilterCol9));
     boolean useBloomFilterCol10 =
         PropertyUtil.propertyAsBoolean(
             table.properties(),
             PARQUET_BLOOM_FILTER_COLUMN_ENABLED_PREFIX + "id_fixed_decimal",
             false);
-    factory.set(
+    writeProperties.put(
         PARQUET_BLOOM_FILTER_COLUMN_ENABLED_PREFIX + "id_fixed_decimal",
         Boolean.toString(useBloomFilterCol10));
     int blockSize =
         PropertyUtil.propertyAsInt(
             table.properties(), PARQUET_ROW_GROUP_SIZE_BYTES, PARQUET_ROW_GROUP_SIZE_BYTES_DEFAULT);
-    factory.set(PARQUET_ROW_GROUP_SIZE_BYTES, Integer.toString(blockSize));
-
-    FileAppender<Record> writer = factory.newAppender(out, format);
-    try (Closeable toClose = writer) {
-      writer.addAll(rows);
+    writeProperties.put(PARQUET_ROW_GROUP_SIZE_BYTES, Integer.toString(blockSize));
+
+    DataWriter<Record> writer =
+        new GenericFileWriterFactory.Builder()
+            .dataSchema(table.schema())
+            .dataFileFormat(format)
+            .writerProperties(writeProperties)
```

Comment:
The sad truth is that this test is not checking if the bloom filters are actually configured/created or not.